nav2_collision_monitor/include/nav2_collision_monitor/polygon.hpp

```diff
@@ -157,6 +158,14 @@ class Polygon
    */
   void publish();
 
+  /**
+   * @brief Set the polygon shape based on the polygon points
+   * @param poly_points Polygon points in the format of x1, y1, x2, y2 ...
+   * @param poly Output polygon points (vertices)
+   * @return True if successfully set the polygon shape
+   */
+  static bool setPolygonShape(std::vector<double> & poly_points, std::vector<Point> & poly);
```

Comment:
This doesn't ever appear to use its staticness?

Reply:
This is used in PolygonVelocity [here](https://github.com/kaichie/navigation2/blob/b97cbb7a490de8c928650ebf74e5cb3f104b1ce5/nav2_collision_monitor/src/velocity_polygon.cpp#L57) and in the polygon class [here](https://github.com/kaichie/navigation2/blob/b97cbb7a490de8c928650ebf74e5cb3f104b1ce5/nav2_collision_monitor/src/polygon.cpp#L434). Or should I create a new `polygon_utils.hpp` for this?

Reply:
Or just add to `utils.hpp` file

Reply:
not relevant anymore in the latest changes.